core/chains/evm/txmgr/evm_tx_store_benchmark_test.go

```diff
@@ -11,161 +10,163 @@ import (
+					ToAddress:      toAddress,
+					EncodedPayload: payload,
+					FeeLimit:       gasLimit,
+					Strategy:       strategy,
+				}, ethClient.ConfiguredChainID())
+				assert.NoError(b, err)
+			}
+
+			b.StopTimer()
+			b.ResetTimer()
+			for n := 0; n < b.N; n++ {
+				toAddress := testutils.NewAddress()
+				b.StartTimer()
+				_, err := txStore.CreateTransaction(tests.Context(b), txmgr.TxRequest{
+					FromAddress:    fromAddress,
+					ToAddress:      toAddress,
+					EncodedPayload: payload,
+					FeeLimit:       gasLimit,
+					Strategy:       strategy,
+				}, ethClient.ConfiguredChainID())
+				b.StopTimer()
+				assert.NoError(b, err)
+			}
+		})
+	}
+}
+
 func BenchmarkFindAttemptsRequiringReceiptFetch(b *testing.B) {
-	db := testutils.NewSqlxDB(b)
-	txStore := NewTestTxStore(b, db)
-	ctx := tests.Context(b)
-
-	blockNum := int64(100)
-
-	kst := cltest.NewKeyStore(b, db)
-	_, fromAddress := cltest.MustInsertRandomKey(b, kst.Eth())
-
-	// Transactions whose attempts should not be picked up for receipt fetch
-	mustInsertFatalErrorEthTx(b, txStore, fromAddress)
-	mustInsertUnstartedTx(b, txStore, fromAddress)
-	mustInsertInProgressEthTxWithAttempt(b, txStore, 4, fromAddress)
-	mustInsertUnconfirmedEthTxWithAttemptState(b, txStore, 3, fromAddress, txmgrtypes.TxAttemptBroadcast)
-	mustInsertConfirmedEthTxWithReceipt(b, txStore, fromAddress, 2, blockNum)
-	// Terminally stuck transaction with receipt should NOT be picked up for receipt fetch
-	stuckTx := mustInsertTerminallyStuckTxWithAttempt(b, txStore, fromAddress, 1, blockNum)
-	mustInsertEthReceipt(b, txStore, blockNum, utils.NewHash(), stuckTx.TxAttempts[0].Hash)
-	// Fatal transactions with nil nonce and stored attempts should NOT be picked up for receipt fetch
-	fatalTxWithAttempt := mustInsertFatalErrorEthTx(b, txStore, fromAddress)
-	attempt := newBroadcastLegacyEthTxAttempt(b, fatalTxWithAttempt.ID)
-	err := txStore.InsertTxAttempt(ctx, &attempt)
-	require.NoError(b, err)
-
-	// Confirmed transaction without receipt should be picked up for receipt fetch
-	confirmedTx := mustInsertConfirmedEthTx(b, txStore, 0, fromAddress)
-	attempt = newBroadcastLegacyEthTxAttempt(b, confirmedTx.ID)
-	err = txStore.InsertTxAttempt(ctx, &attempt)
-	require.NoError(b, err)
-
-	b.ResetTimer()
-	for i := 0; i < b.N; i++ {
-		b.StartTimer()
-		attempts, err := txStore.FindAttemptsRequiringReceiptFetch(ctx, testutils.FixtureChainID)
-		b.StopTimer()
-
-		require.NoError(b, err)
-		require.Len(b, attempts, 1)
-		require.Equal(b, attempt.Hash.String(), attempts[0].Hash.String())
+	for name, size := range benchmarkSizes {
+		b.Run(name, func(b *testing.B) {
+			db := testutils.NewSqlxDB(b)
+			txStore := NewTestTxStore(b, db)
+			ctx := tests.Context(b)
+			blockNum := int64(100)
+			kst := cltest.NewKeyStore(b, db)
+			_, fromAddress := cltest.MustInsertRandomKey(b, kst.Eth())
+
+			for i := 0; i < size; i++ {
+				nonce := evmtypes.Nonce(i * 5) // Generate unique nonces each iteration
```

Comment:
nit: could we just have a nonce external to the loop that we just do nonce++ on every time we use it? Might be easier to maintain if we ever have to revisit this list of tx